tests/hashtable_unittests.cc

```diff
@@ -81,6 +82,12 @@ using namespace testing;
 
 typedef unsigned char uint8;
 
+#if defined _WIN32 || defined __CYGWIN__
+    #ifndef __GNUC__
+      #pragma warning(disable : 4503)
+    #endif
+#endif
+
```

Comment:
It's not "needed", but it keeps the compiler quiet about certain warnings

tests/hashtable_unittests.cc(1952): warning C4503: 'google::sparse_hashtable_const_iterator<Value,Key,HashFcn,ExtractKey,SetKey,EqualKey,Alloc>::sparse_hashtable_const_iterator': decorated name length exceeded, name was truncated
        with
        [
            Value=std::pair<const std::string,google::sparse_hash_map<int,std::string,std::hash<int>,std::equal_to<int>,google::libc_allocator_with_realloc<std::pair<const int,std::string>>>>,
            Key=std::string,
            HashFcn=Hasher,
            ExtractKey=google::sparse_hash_map<std::string,google::sparse_hash_map<int,std::string,std::hash<int>,std::equal_to<int>,google::libc_allocator_with_realloc<std::pair<const int,std::string>>>,Hasher,Hasher,google::libc_allocator_with_realloc<std::pair<const std::string,google::sparse_hash_map<int,std::string,std::hash<int>,std::equal_to<int>,google::libc_allocator_with_realloc<std::pair<const int,std::string>>>>>>::SelectKey,
            SetKey=google::sparse_hash_map<std::string,google::sparse_hash_map<int,std::string,std::hash<int>,std::equal_to<int>,google::libc_allocator_with_realloc<std::pair<const int,std::string>>>,Hasher,Hasher,google::libc_allocator_with_realloc<std::pair<const std::string,google::sparse_hash_map<int,std::string,std::hash<int>,std::equal_to<int>,google::libc_allocator_with_realloc<std::pair<const int,std::string>>>>>>::SetKey,
            EqualKey=Hasher,
            Alloc=google::libc_allocator_with_realloc<std::pair<const std::string,google::sparse_hash_map<int,std::string,std::hash<int>,std::equal_to<int>,google::libc_allocator_with_realloc<std::pair<const int,std::string>>>>>
        ]